app/src/pages/Rankings.tsx

```diff
@@ -0,0 +1,252 @@
+import { useTranslation } from "react-i18next";
+import {
+  Box,
+  Container,
+  Theme,
+  ToggleButton,
+  ToggleButtonGroup,
+  Typography,
+  useMediaQuery,
+} from "@mui/material";
+import * as React from "react";
+import { useEffect, useState } from "react";
+import TextField from "@mui/material/TextField";
+import Autocomplete from "@mui/material/Autocomplete";
+import Stack from "@mui/material/Stack";
+import Switch from "@mui/material/Switch";
+import CircularProgress from "@mui/material/CircularProgress";
+
+import { eventID, Province } from "../components/Types";
+import { getProvinces } from "../components/Provinces";
+import { API_BASE_URL, PRODUCTION } from "../components/api";
+import httpClient from "../httpClient";
+import { RankList } from "../components/RankList";
+import { MyCubingIcon } from "../components/MyCubingIcon";
+
+const provinces: Province[] = getProvinces();
+const events: eventID[] = [
+  "333",
+  "222",
+  "444",
+  "555",
+  "666",
+  "777",
+  "333bf",
+  "333fm",
+  "333oh",
+  "clock",
+  "minx",
+  "pyram",
+  "skewb",
+  "sq1",
+  "444bf",
+  "555bf",
+  "333mbf",
+];
+
+export const Rankings = () => {
+  const { t } = useTranslation();
+  const isSmall = useMediaQuery<Theme>((theme) => theme.breakpoints.down("sm"));
+
+  const [province, setProvince] = useState<Province | null>(provinces[0]);
+  const [eventId, setEventId] = useState<eventID>("333");
+  const [useAverage, setUseAverage] = useState(false);
```

Comment:
Maybe `usingAverage` or something would be better, since hooks start with `use...`.